pennylane/gradients/parameter_shift.py

```diff
@@ -893,7 +897,7 @@ def processing_fn(results):
         shot_vector = isinstance(shots, Sequence)
 
         # analytic derivative of <A>
-        pdA = pdA_fn(results[1:tape_boundary])
+        pdA = pdA_fn(results[int(not pdA_fn.first_result_unshifted) : tape_boundary])

```

Comment:
Here before it was always [1:tape_boundary], now if unshifted we have [1:tape_boundary], and if not unshifted [0:tape_boundary]. So why you need to consider more tape for the not unshifted case?

Reply:
Ah, so the logic is as follows: If the first tape created by the expectation value parameter-shift rule is *not* the unshifted tape (`not pdA_fn.first_result_unshifted = True`), the variance computation requires an unshifted tape and has to add it. In that case, the expectation value results (passed to `pdA_fn`) start with the second entry, the first one is the unshifted one.
If the first tape created by the expectation value parameter-shift rule *is* the unshifted tape (`not pdA_fn.first_result_unshifted = False`), the variance computation does not need to add a tape, so the results passed to `pdA_fn` just start with the very first result :)